src/plugins/core/execute/processedHttpRegionInterceptor.ts

```diff
@@ -0,0 +1,69 @@
+import { HookInterceptor, HookTriggerContext } from 'hookpoint';
+
+import * as models from '../../../models';
+
+export class ProcessedHttpRegionInterceptor implements HookInterceptor<[models.ProcessorContext], boolean | void> {
+  id = 'processedHttpRegion';
+
+  public getResponseLoggingInterceptor(): HookInterceptor<
+    [models.HttpResponse, models.ProcessorContext],
+    boolean | void
+  > {
+    return {
+      id: this.id,
+      afterLoop: this.afterResponseLoggingLoop.bind(this),
+    };
+  }
+
+  public async beforeLoop(
+    hookContext: HookTriggerContext<[models.ProcessorContext], boolean | undefined>
+  ): Promise<boolean | undefined> {
+    const [context] = hookContext.args;
+    if (context.processedHttpRegions) {
+      const processedHttpRegion = this.toProcessedHttpRegion(context);
+      context.processedHttpRegions.push(processedHttpRegion);
+    }
+    return true;
+  }
+
+  public async afterLoop(
+    hookContext: HookTriggerContext<[models.ProcessorContext], boolean | undefined>
+  ): Promise<boolean | undefined> {
+    const [context] = hookContext.args;
+
+    const processedHttpRegion = context.processedHttpRegions?.find(obj => obj.id === context.httpRegion.id);
```

Comment:
wonderful, was wondering how you were going to handle this. New ID field is very useful.

Reply:
However, I find the ID dangerous. It depends on the line in the file and thus is not always unique if the file changes. This is not a problem for this case. but I am slightly afraid that I forget the restriction.